tensorrt_llm/_torch/pyexecutor/py_executor.py

```diff
@@ -873,11 +873,18 @@ def _executor_loop(self):
                 self._pad_attention_dp_dummy_request()
 
                 if self.draft_model_engine is not None or is_ngram:
-                    self._prepare_draft_requests()
+                    self._prepare_draft_requests(self.active_requests)
 
                 scheduled_batch, fitting_disagg_gen_init_requests, num_fitting_reqs = self._schedule(
                 )
 
+                if self.draft_model_engine is not None or is_ngram:
```

Comment:
I think this is not required, since we already added the draft tokens before calling the `schedule`.  It is working correctly when I remove this change. Will add more targeted testing to make sure we don't incorrectly allocate kv cache in this scenario.

CC @pcastonguay / @lfr-0531 if I'm missing anything.